Main/ClientApp/src/Components/Dialogs/DeleteDialog.tsx

```diff
@@ -17,9 +19,12 @@ export default function DeleteDialog(props: DeleteDialogProps) {
       toggleDialogOpen={props.toggleDialogOpen}
       title={props.title}
       description={props.description}
+      additionalCancelAction={props.additionalCancelAction}
       actions={
         <Button
           onClick={() => {
+            props.additionalConfirmAction !== undefined &&
```

Comment:
1. Will this action ever be undefined? It has a default value
2. I don't like splitting the action into two. It should be the responsibility of the caller to define a confirm action, not two ones that are called here subsequently anyway.

Reply:
1. this additionalConfirmAction is optional so if nothing is given it is undefined and cannot be called
2. true, forgot that we provide confirm action in this dialog